test/performance/odp_ipsecfwd.c

```diff
@@ -949,28 +999,337 @@ static odp_bool_t create_sa_dest_queues(odp_ipsec_capability_t *ipsec_capa,
 	return true;
 }
 
-static void create_sa_entry(uint32_t dir, uint32_t spi, const char *src_ip_str,
-			    const char *dst_ip_str, int cipher_idx, uint8_t *cipher_key,
-			    uint8_t *cipher_key_extra, int auth_idx, uint8_t *auth_key,
-			    uint8_t *auth_key_extra, uint32_t icv_len, uint32_t ar_ws,
-			    uint32_t max_num_sa, prog_config_t *config)
+static odp_bool_t lookup_int(config_t *cfg, const char *root, const char *entry, int *val)
 {
-	uint32_t src_ip, dst_ip;
-	odp_ipsec_sa_param_t sa_param;
-	odp_ipsec_crypto_param_t crypto_param;
+	char path[MAX_PATH_LEN];
+	int tmp;
+
+	memset(path, 0, sizeof(path));
+	strcat(strcat(path, root), entry);
+
+	if (config_lookup_int(cfg, path, &tmp) == CONFIG_TRUE) {
+		*val = tmp;
+		return true;
+	}
+
+	return false;
+}
+
+static odp_bool_t lookup_int64(config_t *cfg, const char *root, const char *entry,
+			       long long *val)
+{
+	char path[MAX_PATH_LEN];
+	long long tmp;
+
+	memset(path, 0, sizeof(path));
+	strcat(strcat(path, root), entry);
+
+	if (config_lookup_int64(cfg, path, &tmp) == CONFIG_TRUE) {
+		*val = tmp;
+		return true;
+	}
+
+	return false;
+}
+
+static odp_bool_t lookup_string(config_t *cfg, const char *root, const char *entry, char *val)
+{
+	char path[MAX_PATH_LEN];
+	const char *tmp;
+
+	memset(path, 0, sizeof(path));
+	strcat(strcat(path, root), entry);
+
+	if (config_lookup_string(cfg, path, &tmp) == CONFIG_TRUE) {
+		strcpy(val, tmp);
+		return true;
+	}
+
+	return false;
+}
+
+static void parse_sa_entry(config_t *cfg, const char *root, sa_config_t *config)
+{
+	int val;
+	char val_str[16U];
+
+	if (lookup_int(cfg, root, "sa_dir", (int *)&config->sa_param.dir))
```

Comment:
This function assumes that the size of the enums is the same of int, which is typically true but not always. It would be nice to not make the assumption.

Reply:
Perhaps safer to handle them via a helper, changed in v3.